contracts/forwarding/astroport-forwarding/src/reply.rs

```diff
@@ -0,0 +1,44 @@
+use cosmwasm_std::{ensure, DepsMut, Env, Reply, Response, SubMsgResult};
+use euclid::error::ContractError;
+
+use crate::state::FORWARDING_STATE;
+
+pub const ASTRO_SWAP_REPLY_ID: u64 = 1;
+
+pub fn on_astro_swap_reply(deps: DepsMut, env: Env, msg: Reply) -> Result<Response, ContractError> {
+    match msg.result.clone() {
+        SubMsgResult::Err(err) => Err(ContractError::new(&format!(
```

Comment:
No changes are made to the `FORWARDING_STATE` in case the swap returns an error?

Reply:
Yeah i think the funds are handled, but what about the state after an error in the reply? Won't the contract get stuck at this?
```rust
    ensure!(
        FORWARDING_STATE.may_load(deps.storage)?.is_none(),
        ContractError::new("Contract already in forwarding state")
    );
```

Reply:
Also maybe we could make the above check earlier, possible at the beginning of the `ExecuteMsg` entry point?